src/Text/Markdown/SlamDown/Parser/Inline.purs

```diff
@@ -315,7 +315,7 @@ inlines = many inline2 <* eof
     textBox = do
       ty <- parseTemplate
       skipSpaces
-      def <- optionMaybe $ parens $ expr id parseValue
+      def <- optionMaybe $ parens $ expr id $ parseValueOfType ty
       pure $ TextBox ty def
 
     und :: Parser String String
```

Comment:
this resolves some ambiguities in the grammar, but now we can't quickcheck the parser with the current `Arbitrary` instances; that is, we're just generating random strings for default values, but we need to be generating strings that correspond to the right text box type.

This is all so gnarly and horrible; it might be easier to figure out how to fix this if we added a minimal ADT/grammar of value/constant expressions concretely, and then required that the user provide an injection from their AST into this one, rather than use `String` everywhere by default. But this may be introducing too much complexity for the common case, I don't know.